src/Schema/Blueprint.php

```diff
@@ -34,4 +36,26 @@ public function toSql(Connection $connection, Grammar $grammar)
 
         return $this->creating() ? $this->inlineCreateIndexStatements($statements) : $statements;
     }
+
+    /**
+     * Execute the blueprint against the database.
+     *
+     * @param  \Illuminate\Database\Connection  $connection
+     * @param  \Illuminate\Database\Schema\Grammars\Grammar  $grammar
+     * @return void
+     */
+    public function build(Connection $connection, Grammar $grammar)
+    {
+        try {
+            foreach ($this->toSql($connection, $grammar) as $statement) {
+                $connection->statement($statement);
+            }
```

Comment:
rather than re-implementing build, can we instead call BaseBlueprint's implementation of build and just wrap it in a try/catch?

Reply:
Bumping this question

Reply:
Totally, completely forgot that was possible. Much better solution